NitroxServer/Serialization/BatchCellsParser.cs

```diff
@@ -1,128 +1,207 @@
+using NitroxServer.UnityStubs;
+
+namespace NitroxServer.Serialization
+{
+    /**
+     * Parses the files in build18 in the format of batch-cells-x-y-z-slot-type.bin
+     * These files contain serialized GameObjects with EntitySlot components. These
+     * represent areas that entities (creatures, objects) can spawn within the world.
+     * This class consolidates the gameObject, entitySlot, and cellHeader data to
+     * create EntitySpawnPoint objects.
+     */
+    class BatchCellsParser
+    {
+        public bool WarnSilence;
+        public bool DoNotAttempt;
+        private readonly ServerProtobufSerializer serializer;
+        private readonly Dictionary<string, Type> surrogateTypes = new Dictionary<string, Type>();
+
+        public void CheckForIndexAndSilenceWarnings()
+        {
+            /**
+           * Implemented check to see if index.txt in our Build18 folder exists.
+           * If it does, warnings for this check are then silenced on the console, and instead, missing files are dumped to a missing.txt file
+           * Warnings can be turned back on by the presence of a nosilence.txt file. Could be later implemented as an option. */
+            if (WarnSilence)
+            {
+                return;
+            }
+            if (DoNotAttempt)
+            {
+                return;
+            }
+            string PathToSub;
+            List<string> errors = new List<string>();
+            Optional<string> subnauticaPath = GameInstallationFinder.Instance.FindGame(errors);
+            if (subnauticaPath.IsEmpty())
+            {
+                string pathTXTFile = Path.Combine(Path.GetFullPath("."), "path.txt");
+                Log.Info("Using " + pathTXTFile + " as path to path.txt");
+                PathToSub = File.ReadAllText(pathTXTFile);
+            } else
+            {
+                PathToSub = subnauticaPath.Get();
+            }
+            if (PathToSub.Length <= 1)
+            {
+                Log.Warn("Empty Path.txt file or universal finder can't find game directory! Can not continue executing method!");
+                return;
+            }
+            string fileName = Path.Combine(PathToSub, "SNUnmanagedData\\Build18\\index.txt");
+            Log.Info("Checking for " + fileName);
+            if (!File.Exists(fileName))
+            {
+                Log.Warn("Can not find index.txt!");
+                return;
+            }
+            fileName = Path.Combine(".\\", "nosilence.txt");
+            if (!File.Exists(fileName))
+            {
+                Log.Info("Index found! Warnings about missing files are silenced and missing filenames are now dumped to missing.txt!");
+                WarnSilence = true;
+            }
+            else
+            {
+                Log.Info("Not silencing warnings about missing files!");
+                DoNotAttempt = true;
+            }
+            return;
+        }
+        public void LogMissingFileName(string fileName)

```

Comment:
Missing some of these files is perfectly normal as not every batch has corresponding files.  To take care of the over logging we need to bundle a release version that has debug logging disabled.  Then no one would be confused by the error messages.

Reply:
Yeah I think we should close without merge as a recent commit has reduced this logging.  Thanks for submitting though - we look forward to future PRs :)